platform/openide.dialogs/src/org/openide/NotifyDescriptor.java

```diff
@@ -1153,4 +1163,209 @@ JTextField createTextField() {
         }
     }
      // end of InputLine
+    
+    /** Notification providing a selection list allowing multiple selections.
+    * @since 7.60
+    */
+    public static final class QuickPick extends NotifyDescriptor {
+
+        private final Map<JToggleButton, Item> btn2items = new LinkedHashMap<JToggleButton, Item>();
+        private final boolean multipleSelection;
+
+        /**
+         * Construct dialog with the specified title and label text.
+         * @param text label text
+         * @param title title of the dialog
+         * @param items a list of items
+         * @param multipleSelection true if multiple selection allowed
+         * @since 7.60
+         */
+        public QuickPick(final String text, final String title, final List<Item> items, final boolean multipleSelection) {
+            super(null, title, OK_CANCEL_OPTION, PLAIN_MESSAGE, null, null);
+            this.multipleSelection = multipleSelection;
+            super.setMessage(createDesign(text, items));
```

Comment:
Now, when we know we also want to display this in LSP - wouldn't it be better to delay creation of the Swing component until it is really needed?

Reply:
Fixed.